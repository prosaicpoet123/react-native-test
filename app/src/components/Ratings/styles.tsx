import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  ratings: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
})

export default styles
