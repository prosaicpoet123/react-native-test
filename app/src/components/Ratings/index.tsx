import React, { FC } from 'react'
import { Text, View } from 'react-native'
import { IProduct } from '../../@Types/product'
import { Rating } from 'react-native-ratings'
import styles from './styles'

interface IRatings {
  count: IProduct['review_count']
  rating: IProduct['rating']
}

type IProps = IRatings

const Ratings: FC<IProps> = ({ rating, count }) => (
  <View style={styles.ratings}>
    <Rating startingValue={rating} readonly imageSize={20} />
    <Text> ({count})</Text>
  </View>
)

export default Ratings
