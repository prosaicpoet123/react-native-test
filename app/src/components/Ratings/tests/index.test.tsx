import React from 'react'
import renderer from 'react-test-renderer'
import Ratings from '../'

describe('<Ratings />', () => {
  it('renders correctly', () => {
    const ratings = { rating: 3.5, count: 20 }
    const tree = renderer.create(<Ratings {...ratings} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
