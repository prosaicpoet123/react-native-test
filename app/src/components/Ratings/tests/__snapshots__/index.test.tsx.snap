// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ratings /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "flex-start",
    }
  }
>
  3.5
  <Text>
     (
    20
    )
  </Text>
</View>
`;
